java-frontend/src/test/files/se/BinaryExpressionNotNull.java

```diff
@@ -0,0 +1,16 @@
+class A {
+
+  void test(Object o) {
+    Boolean b = o != null;
+    if (b == null) { // Noncompliant {{Change this condition so that it does not always evaluate to "false"}}
+
+    }
+
+    String s1 = null;
+    String s2 = "test";
+    String s = s1 + s2;
```

Comment:
Can you add a comment here that a NPE is expected? with a link to MMF-859.

Reply:
String concatenation will not fail with null value, result will be `"nulltest"` ,  but good point, I will add another case using `Integer`